warehouse/query-core/src/main/java/datawave/query/jexl/nodes/QueryPropertyMarker.java

```diff
@@ -59,18 +59,22 @@ public static String label() {
     }
     
     /**
-     * Create and return a new query property marker of the supplied marker type with the specified source via reflection.
+     * Create and return a new query property marker of the supplied marker type with the specified source via reflection. If the source node or one of its
+     * ancestors is already marked with the supplied marker type the original source node is returned instead of a new marker instance.
      * 
      * @param source
      *            the source node
      * @param markerType
      *            the marker type
      * @param <MARKER>
      *            the marker type
-     * @return the new query property marker instance
+     * @return the new query property marker instance, or the original source if the marker has already been applied
      */
-    public static <MARKER extends QueryPropertyMarker> MARKER create(JexlNode source, Class<MARKER> markerType) {
+    public static <MARKER extends QueryPropertyMarker> JexlNode create(JexlNode source, Class<MARKER> markerType) {
         try {
+            if (isSourceMarked(source, markerType) || isAncestorMarked(source, markerType)) {
+                return source;
```

Comment:
should this return the passed source or the source of the QueryPropertyMarker? They may not be the same.

Reply:
It should return the original node that was passed in. This keeps things simple for the caller.